mne/io/neuralynx/neuralynx.py

```diff
@@ -101,6 +102,56 @@ def __init__(
             sfreq=nlx_reader.get_signal_sampling_rate(),
         )
 
+        ncs_fnames = nlx_reader.ncs_filenames.values()
+        ncs_hdrs = [
+            hdr
+            for hdr_key, hdr in nlx_reader.file_headers.items()
+            if hdr_key in ncs_fnames
+        ]
+
+        # if all files have the same recording_opened date, write it to info
+        meas_dates = np.array([hdr["recording_opened"] for hdr in ncs_hdrs])
+        # to be sure, only write if all dates are the same
+        write_meas = (meas_dates == meas_dates[0]).all()
+        if not write_meas:
+            logger.warning(
+                "Not all .ncs files have the same recording_opened date. "
+                + "Not writing meas_date to info."
+            )
```

Comment:
Maybe instead of this you could just use the meas date from the first file if that's well-defined?